File: ruby/phone-number/phone_number.rb
class PhoneNumber
  def self.clean(num)
    phone = num.gsub(/\D/, '').gsub(/^1/, '')
    return nil unless (phone.length == 10) && phone.match(/^[2-9]..[2-9]/)
    phone
  end
end

module BookKeeping
  VERSION = 2
end

File: javascript/phone-number/phone-number.js
class PhoneNumber {
  constructor(num) {
    this.num = num;
  }

  number() {
    const phone = this.num.replace(/\D/g, '').replace(/^1/, '');
    if ((phone.length !== 10) || !(/^[2-9]..[2-9]/).test(phone)) {
      return null;
    }
    return phone;
  }
}

module.exports = PhoneNumber;
